packages/@aws-cdk/aws-cloudfront/lib/function.ts

```diff
@@ -0,0 +1,109 @@
+import { IResource, Names, Resource } from '@aws-cdk/core';
+import { Construct } from 'constructs';
+import { CfnFunction } from './cloudfront.generated';
+
+/**
+ * Represents the function's source code
+ */
+export abstract class FunctionCode {
+
+  /**
+   * Inline code for function
+   * @returns `InlineCode` with inline code.
+   * @param code The actual function code
+   */
+  public static fromInline(code: string): InlineCode {
+    return new InlineCode(code);
+  }
+
+  /**
+   * renders the function code
+   */
+  public abstract render(): string;
+}
+
+export class InlineCode extends FunctionCode {
```

Comment:
Is there any reason to export this? If we change the signature of `fromInline` to just return `FunctionCode`, then this could be left un-exported, correct?

Reply:
Still think we should default to leaving this un-exported until/unless we have a good reason to export it.